torchaudio/functional/filtering.py

```diff
@@ -938,20 +941,31 @@ def lfilter(
 
     Args:
         waveform (Tensor): audio waveform of dimension of ``(..., time)``.  Must be normalized to -1 to 1.
-        a_coeffs (Tensor): denominator coefficients of difference equation of dimension of ``(n_order + 1)``.
+        a_coeffs (Tensor): denominator coefficients of difference equation of dimension of ``(*, n_order + 1)``.
+                                Where * is the optional number of filter banks.
                                 Lower delays coefficients are first, e.g. ``[a0, a1, a2, ...]``.
                                 Must be same size as b_coeffs (pad with 0's as necessary).
-        b_coeffs (Tensor): numerator coefficients of difference equation of dimension of ``(n_order + 1)``.
-                                 Lower delays coefficients are first, e.g. ``[b0, b1, b2, ...]``.
-                                 Must be same size as a_coeffs (pad with 0's as necessary).
+        b_coeffs (Tensor): numerator coefficients of difference equation of dimension of ``(*, n_order + 1)``.
+                                Where * is the optional number of filter banks.
+                                Lower delays coefficients are first, e.g. ``[b0, b1, b2, ...]``.
+                                Must be same size as a_coeffs (pad with 0's as necessary).
         clamp (bool, optional): If ``True``, clamp the output signal to be in the range [-1, 1] (Default: ``True``)
 
     Returns:
-        Tensor: Waveform with dimension of ``(..., time)``.
+        Tensor: Waveform with dimension of ``(..., *, time)``.
     """
     # pack batch
     shape = waveform.size()
-    waveform = waveform.reshape(-1, shape[-1])
+    assert a_coeffs.size() == b_coeffs.size()
+    assert a_coeffs.ndim <= 2
+
+    waveform = waveform.reshape(-1, 1, shape[-1])
+    if a_coeffs.ndim > 1:
+        shape = shape[:-1] + (a_coeffs.shape[0], shape[-1])
+        waveform = waveform.repeat(1, a_coeffs.shape[0], 1)
```

Comment:
Instead of repeating the waveform manually _and_ computing the shape manually, how about overwriting `shape` after repeating? I think that will be slightly easier to understand.

```python
waveform = waveform.repeat(...)
shape = waveform.size()
```